lib/logflare/utils/debugging.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Logflare.Utils.Debugging do
+  @moduledoc false
+  alias Logflare.Backends.IngestEventQueue
+
+  def list_counts(source_id) do
+    :erpc.multicall(
+      [Node.self() | Node.list()],
+      fn ->
+        {Node.self(), IngestEventQueue.list_counts({source_id, nil})}
+      end,
```

Comment:
How do you perform deployments? Do you keep the old cluster and new cluster together for a while to migrate that or fully separate?

The reason I ask is because anonymous functions do not work if could have mixed versions, then in this case, you should do something like this instead:

```elixir
  def list_counts(source_id) do
    :erpc.multicall(
      [Node.self() | Node.list()], __MODULE__, list_counts_callback, [source_id]
    )
  end

  def list_counts_callback(source_id) do
    {Node.self(), IngestEventQueue.list_counts({source_id, nil})}
  end
```

I also understand this is debug only but perhaps you want to debug counts during deployments and if they are mixed clusters, then you wouldn't be able to :)

Reply:
yea right now there is a chance for that to happen with rolling deployments.
we do a mix of rolling  and green-blue depending on risk tolerance for that specific cluster.
Adjusted, thanks! 💟